app/services/publisher_balance_getter.rb

```diff
@@ -56,6 +32,45 @@ def channels_query_string
     publisher.channels.verified.map { |channel| "&account=#{URI.escape(channel.details.channel_identifier)}" }.reduce(:+)
   end
 
+  # Eyeshade may return a 0 balance or an empty response for accounts (channel and owner)
+  # with no balance, so we must fill in these values
+  def fill_in_missing_accounts(accounts)
+    # Find all of the publisher's verified channel ids
+    verified_channel_ids = publisher.channels.verified.map { |channel| channel.details.channel_identifier }
+
+    # Find which verified channel ids have been included in the response
+    account_ids = accounts.map { |account| account["account_id"] }
+
+    # Find whether the owner account was included in the response
+    accounts_include_owner_account = false
+    accounts.each do |account|
+      if account["account_type"] == "owner"
+        accounts_include_owner_account = true
+      end
+    end
+
+    # Fill in missing channel accounts if needed
+    channel_ids_to_fill_in = (verified_channel_ids.map { |verified_channel_id| account_ids.include?(verified_channel_id) ? nil : verified_channel_id }).compact
```

Comment:
I think you can do `verified_channel_ids.select { |verified_channel_id| account_ids.include?(verified_channel_id) }`